test/battle/hold_effect/jaboca_berry.c

```diff
@@ -4,41 +4,57 @@
 ASSUMPTIONS
 {
     ASSUME(gItems[ITEM_JABOCA_BERRY].holdEffect == HOLD_EFFECT_JABOCA_BERRY);
+    ASSUME(gBattleMoves[MOVE_TACKLE].split == SPLIT_PHYSICAL);
 }
 
 SINGLE_BATTLE_TEST("Jaboca Berry causes the attacker to lose 1/8 of its max HP if a physical move was used")
 {
     s16 damage;
+    u16 move;
+
+    PARAMETRIZE { move = MOVE_SWIFT; }
+    PARAMETRIZE { move = MOVE_TACKLE; }
 
     GIVEN {
-        ASSUME(gBattleMoves[MOVE_TACKLE].split == SPLIT_PHYSICAL);
+        ASSUME(gBattleMoves[MOVE_SWIFT].split == SPLIT_SPECIAL);
         PLAYER(SPECIES_WOBBUFFET);
         OPPONENT(SPECIES_WOBBUFFET) { Item(ITEM_JABOCA_BERRY); }
     } WHEN {
-        TURN { MOVE(player, MOVE_TACKLE); }
+        TURN { MOVE(player, move); }
     } SCENE {
-        ANIMATION(ANIM_TYPE_MOVE, MOVE_TACKLE, player);
+        ANIMATION(ANIM_TYPE_MOVE, move, player);
         HP_BAR(opponent);
-        ANIMATION(ANIM_TYPE_GENERAL, B_ANIM_HELD_ITEM_EFFECT, opponent);
-        HP_BAR(player, captureDamage: &damage);
-        MESSAGE("Wobbuffet was hurt by Foe Wobbuffet's Jaboca Berry!");
+        if (move == MOVE_TACKLE) {
+            ANIMATION(ANIM_TYPE_GENERAL, B_ANIM_HELD_ITEM_EFFECT, opponent);
+            HP_BAR(player, captureDamage: &damage);
+            MESSAGE("Wobbuffet was hurt by Foe Wobbuffet's Jaboca Berry!");
+        } else {
+            NONE_OF {
+                ANIMATION(ANIM_TYPE_GENERAL, B_ANIM_HELD_ITEM_EFFECT, opponent);
+                MESSAGE("Wobbuffet was hurt by Foe Wobbuffet's Jaboca Berry!");
+            }
+        }
     } THEN {
-        EXPECT_EQ(player->maxHP / 8, damage);
+        if (move == MOVE_TACKLE)
+            EXPECT_EQ(player->maxHP / 8, damage);
     }
 }
 
-SINGLE_BATTLE_TEST("Jaboca Berry is not triggered by a special move")
+SINGLE_BATTLE_TEST("Jaboca Berry tirggers before Bug Bite can steal it")
```

Comment:
Not sure what to do here since Bulbapedia and Showdown give conflicting information. Would be good if someone tested it in-game.

Reply:
For the record, we found that Showdown was wrong.
https://github.com/smogon/pokemon-showdown/issues/9853